src/lupapiste_commons/i18n/txt_resources.clj

```diff
@@ -0,0 +1,99 @@
+(ns lupapiste-commons.i18n.txt-resources
+  (:require [clojure.edn :as edn]
+            [clojure.java.io :as io]
+            [clojure.string :as s]
+            [flatland.ordered.map :refer [ordered-map]])
+  (:import [java.io File PushbackReader StringReader]
+           [java.net URL]))
+
+(set! *warn-on-reflection* true)
+
+(defn source-changed? [sym]
+  (:source-changed (meta sym)))
+
+(defn write-key [sym]
+  (cond-> (str sym)
+    (source-changed? sym) (str "!")))
+
+(defn nil->empty-str [x]
+  (if (nil? x) "" x))
+
+(defn read-key [^String string]
+  (if (.endsWith string "!")
+    (-> (s/replace string #"!$" "")
+        symbol
+        (with-meta {:source-changed true}))
+    (symbol string)))
```

Comment:
Tämähän on sinällään vanhaa koodia, mutta koska vähän ihmettelin tätä jo silloin kuin Docjuren kanssa puljasin: onkohan tällä symbol - meta -pyörityksellä jokin syvempi idea? Kun jotenkin kävi aikoinaan mielessä sellainen kerettiläinen ajatus, että kyllä on tekstidatan käsittely tehty vaikeaksi 😃  

No, ehkä tämä on tavallaan näppärä tapa tuon huutomerkin kuljetukseen 🤷

Reply:
Enpä kyllä keksi mikä pointti. Näin tämä on näköjään ollut alusta asti vuodesta 2015, mutta en löytänyt todisteita, että noissa avaimissa olisi huutomerkkejä ollut, eli en tiedä miksi ne poistetaan. Enkä tiedä miksi näistä tehdään symboleja tässä. Mutta eipä tätä nyt henno muuttaakaan, jos vaikka jokin jossain tätä käyttää.